src/interceptors/exception.interceptor.ts

```diff
@@ -0,0 +1,35 @@
+import {
+  ArgumentsHost,
+  Catch,
+  ExceptionFilter,
+  HttpException,
+  Logger,
+} from '@nestjs/common';
+import { Response } from 'express';
+import { APIResponse } from 'src/utils/apiResponse.utils';
+
+@Catch(HttpException)
+export class HttpExceptionFilter implements ExceptionFilter {
+  private logger = new Logger();
+
+  catch(exception: HttpException, host: ArgumentsHost) {
+    let err = exception['response'];
+    err = err.response ? err.response : JSON.stringify(err);
+    const context = host.switchToHttp();
+    const response = context.getResponse<Response>();
+    const status = exception.getStatus();
+    this.logger.error(
+      `Message : ${err}`,
+      `${context.getRequest().method} ${context.getRequest().url}`,
+    );
+    const error =
+      typeof exception.getResponse() === 'string'
+        ? exception.getResponse()
+        : exception.getResponse() &&
+            typeof exception.getResponse()['message'] === 'object'
+          ? exception.getResponse()['message'].join(', ')
+          : exception.getResponse()['message'];
```

Comment:
PullProofAI

The error handling logic can be improved by simplifying the conditional checks and extracting them into separate functions for better readability and maintainability.
```diff
+ // Extract error response handling logic into a separate function for clarity
+ // Simplify the conditional checks for error response handling
```

33:
Ensure that the `APIResponse.create` method is properly documented and handles all possible cases to avoid unexpected behavior.
```diff
+ // Ensure APIResponse.create method is well-documented and handles all cases
```

### Overall, the exception interceptor code looks well-structured and handles HTTP exceptions effectively. Consider adding comments for clarity and improving the error handling logic for better readability. 



<!-- This is an auto-generated comment by PullProofAI -->